### ISSUE TITLE
Vaccination certificate 1/2 detailed view changes background color

### ISSUE THREAD

**Issue by MikeMcC399** [corona-warn-app/cwa-app-android#3927]
## Avoid duplicates
* [X] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [X] Bug is specific for Android only, according to https://github.com/corona-warn-app/cwa-app-android/issues/3926#issuecomment-898903846 from @dsarkar 
* [ ] Bug is not already reported in another issue
Issue is mentioned also in https://github.com/corona-warn-app/cwa-app-android/issues/3926

## Technical details

- Device name: Samsung Galaxy A5 (2017) SM-A520F
- Android version: 8.0.0
- App version: 2.6.1 (also 2.7.1-RC0 and 2.8.0-RC0)

## Describe the bug

The detail view of a digital COVID vaccination certificate where the certificate is not the final one of a series is shown with a grey background, but only if the final vaccination certificate is not also stored on the device.

## Steps to reproduce the issue

1. Remove any existing certificates from the device
2. Tap "Certificates" tab, tap "+ CERTIFICATE"
3. Scan in a 1/2 vaccination certificate
4. Note that it is shown with a grey background
5. Tap back arrow, tap "X"
6. Tap "+ CERTIFICATE"
7. Scan in a 2/2 vaccination certificate (vaccination more than 14 days ago)
8. Note that it is shown with a blue background
9. Tap back arrow
10. Scroll down to Vaccination 1 of 2 and tap it
11. Note that it is shown with blue background
12. Tap back arrow
13. Tap Vaccination 2 of 2
14. Tap three-dot symbol, tap "Remove" and tap "REMOVE"
15. Tap Vaccination 1 of 2
16. Note that now it is shown with grey background, which is different from in step 11

This is an example, created by scanning in https://www.coronawarn.app/assets/screenshots/2-6/en/ios/iPhone%2011%20Pro-screenshot_first_vaccination_certificate_details_part1.png:

![VaccinationDetailsFragment_2 6 1](https://user-images.githubusercontent.com/66998419/129439402-9a17fdf2-223b-4b6b-9949-af32b412ba8b.png)

## Expected behavior

The background color for the detail display of a non-final vaccination certificate should not change depending on whether the final vaccination certificate is also stored on the device.

Implicitly the detail display of one certificate is only about that one certificate being displayed and the display should not depend on other states on the device, like the characteristics of an additional certificate also stored on the device.

## Possible Fix

Make the background color of a vaccination certificate in the certificate detailed view depend only on the characteristics of the certificate itself.

**Comment by Jo-Achim** [comment]
@MikeMcC399 
> Make the background color of a vaccination certificate in the certificate detailed view depend only on the characteristics of the certificate itself.
>
If I understand correctly...
This means, the background color of certificate (**1/2**) - non-final vaccination - _always should be gray_ and the background color of certificate (**2/2**) - final vaccination - _always should be blue_.
The background color of certificate (**1/1**)  - final vaccination - _always should be blue_, too.
I think that would be correct.

**Comment by MikeMcC399** [comment]
@Jo-Achim 
I think what you describe is correct. The background color should be linked to the QR code and certificate on display only.

Also the behavior on Android and iOS should be made consistent.

One interesting side effect of the background color / gradient not being linked solely to the certificate being displayed is that if you have one mobile device loaded with a 1/2 and a 2/2 certificate, then you display the 1/2 certificate it shows with a blue background, BUT if you scan the QR code of the 1/2 certificate from the display where it is showing a blue background, scanning using a second device with no previous certificates stored, then on the second device the 1/2 certificate shows with a grey background.

This is confusing and inconsistent.

**Comment by Jo-Achim** [comment]
@MikeMcC399 

Yes, I think that clear algorithms should already be deposited here.

However, I would like to consider the following restriction:
Against the background that it should only be a question of time when a 3rd vaccination belongs to 'completeness' (a third vaccine where previously 2 vaccines belong to the "complete vaccination" or a second vaccine where previously only one vaccination leads to "completeness"), the fact should also be taken into account that a previously 'blue' certificate ((**2/2**) or (**1/1**)) might have to become 'gray' if the third / second Vaccination certificate is added.

In the future, this can be extended to other necessary vaccinations to achieve a 'complete vaccination'.

**Comment by MikeMcC399** [comment]
@Jo-Achim 
> However, I would like to consider the following restriction:
Against the background that it should only be a question of time when a 3rd vaccination belongs to 'completeness' (a third vaccine where previously 2 vaccines belong to the "complete vaccination" or a second vaccine where previously only one vaccination leads to "completeness"), the fact should also be taken into account that a previously 'blue' certificate ((2/2) or (1/1)) might have to become 'gray' if the third / second Vaccination certificate is added.

That is probably a different topic, even if it is related given the way the Android app is currently programmed.
A 2/2 certificate will still be complete in its own right until it expires (currently in Germany after one year from date of vaccination - see https://www.coronawarn.app/en/faq/#eu_dcc_validity) , even if a refresher is added. The Business Rules for Germany could of course be changed over time. I haven't seen any statement yet how a refresher vaccination will be coded though.
